pico/services/src/main/java/io/helidon/pico/services/Utils.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.pico.services;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import io.helidon.pico.Application;
+import io.helidon.pico.DependenciesInfo;
+import io.helidon.pico.ServiceInfo;
+import io.helidon.pico.ServiceProvider;
+
+/**
+ * Public helpers around shared Pico services usages.
+ */
+public class Utils {
```

Comment:
Public utility classes should not be called `Utils`. We can have public helpers, but with some reasonable naming.

Reply:
Each module has a general Utils for that module. What do you suggest they be named canonically?

Reply:
I will presume `ServicesUtils` for this case, and do the same for the others.